openedx/openedx-platform
# feat: add user_id in the default scopes if no scopes are requested from payload

## Description

- This change will impact API users with Application Access and grant type of `client_credentials`.
- Defined scopes in the ApplicationAccess were not being utilized if request payload didn't have `scope` attribute. Only default scopes were being used.
- Now `user_id` will be included in the default scopes if no scopes are requested and `user_id` is an allowed scope.

## Supporting information

Ticket: [ENT-8426: Let API users get a user_id in their JWTs](https://2u-internal.atlassian.net/browse/ENT-8426)

## Testing instructions

1. Create an Application for a user with a grant type of client credentials.
2. Create an ApplicationAccess record for that Application and add its scope attributes, which are delimited by a comma.
3. Request access token from the API, and it should include all attributes defined in the Application scopes.
